### ISSUE TITLE
Mismatch between logits dimensions and vocab size in ESM C model

### ISSUE THREAD

**Issue by navotarad** [evolutionaryscale/esm#281]
I am running the ESM C quickstart example (shown in the top-level README).
The code runs without error but the dimensions of the logits output are not what I would expect.

I assumed that the last dimension of the logits tensor would be the same size as the vocabulary specified in the tokenizer,
so that if I take an argmax over that last dimension I would get encoded tokens back, but this is not the case.
Instead the last output dimension has size 64 (with a vocab size of 33)

The size 64 dimension comes from a hard-coded value in the ESM C model definition https://github.com/evolutionaryscale/esm/blob/23b084b94f3b5f6fd700ae41f46a2787a9a5c12b/esm/models/esmc.py#L75

Am I mis-interpreting what the logits output represents?
If so how should I interpret the logits, and is there a way to get from the logits output back to encoded tokens?

Minimal reproducible example with:
- Python version: 3.12.11
- esm version: 3.1.6
```python
from esm.models.esmc import ESMC
from esm.sdk.api import ESMProtein, LogitsConfig

protein = ESMProtein(sequence="AAAAA")
client = ESMC.from_pretrained("esmc_300m").to("cpu")
protein_tensor = client.encode(protein)
logits_output = client.logits(
   protein_tensor, LogitsConfig(sequence=True)
)

print(len(client.tokenizer)) # 33
print(logits_output.logits.sequence.shape) # actual (1, 7, 64), expected (1, 7, 33)

**Comment by imathur1** [comment]
For ESM3 and ESMC we use 64-dimensional logits for compatibility with transformer engine. Indices 0 to 32 out of 63 correspond to the tokens in the vocab, and the rest have no meaning. I hope this helps!

**Comment by navotarad** [comment]
That does help. Thanks for clearing it up
